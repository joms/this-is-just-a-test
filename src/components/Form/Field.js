import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Field = ({ children, horizontal }) => <div className={cx('field', { horizontal })}>{children}</div>;

Field.defaultProps = {
    horizontal: false,
};

Field.propTypes = {
    children: PropTypes.node.isRequired,
    horizontal: PropTypes.bool,
};

export default Field;
