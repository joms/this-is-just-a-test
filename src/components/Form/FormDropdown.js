import React from 'react';
import { Field } from '.';
import Dropdown from '../Dropdown';

const FormDropdown = ({ ...args }) => (
    <Field>
        <Dropdown {...args} />
    </Field>
);

FormDropdown.propTypes = Dropdown.propTypes;

export default FormDropdown;
