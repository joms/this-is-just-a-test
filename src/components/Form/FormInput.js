import React from 'react';
import { Field } from '.';
import Input from '../Input';

const FormInput = ({ ...args }) => (
    <Field>
        <Input {...args} />
    </Field>
);

FormInput.propTypes = Input.propTypes;

export default FormInput;
