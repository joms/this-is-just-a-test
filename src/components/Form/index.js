import Form from './Form';
import Field from './Field';

export { Form, Field };

export default Form;
