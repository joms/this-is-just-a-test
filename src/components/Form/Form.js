import React from 'react';
import PropTypes from 'prop-types';

import Field from './Field';
import FormInput from './FormInput';
import FormDropdown from './FormDropdown';

const Form = ({ children, onSubmit }) => {
    const handleSubmit = (e, ...args) => {
        e.preventDefault();
        onSubmit(e, args);
    };

    return <form onSubmit={handleSubmit}>{children}</form>;
};

Form.defaultProps = {
    handleSubmit: () => null,
};

Form.propTypes = {
    children: PropTypes.node.isRequired,
    handleSubmit: PropTypes.func,
};

Form.Field = Field;
Form.Input = FormInput;
Form.Dropdown = FormDropdown;

export default Form;
