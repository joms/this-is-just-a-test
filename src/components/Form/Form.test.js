import React from 'react';
import { shallow } from 'enzyme';
import Form from '.';

describe('<Form>', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<Form onSubmit={onSubmit}>I am a form!</Form>);
    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('submits', () => {
        wrapper.simulate('submit', { preventDefault: () => null });
        expect(onSubmit).toHaveBeenCalled();
    });
});
