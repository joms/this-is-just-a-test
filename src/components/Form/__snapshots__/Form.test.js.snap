// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  I am a form!
</form>
`;
